surveymonkey/surveymonkey.py

```diff
@@ -115,16 +168,30 @@ class SurveyMonkeyXBlock(XBlock, StudioEditableXBlockMixin):
         "weight",
         "client_id",
         "client_secret",
+        "inline_survey_view",
+        "overwrite_survey_questions",
+        "previous_survey_name",
+        "overwritten_question_headings",
+        "surveymonkey_api_cache_duration",
     )
 
     def validate_field_data(self, validation, data):
         try:
-            api_survey_monkey = ApiSurveyMonkey(data.client_id, data.client_secret)
+            api_survey_monkey = ApiSurveyMonkey(
+                data.client_id,
+                data.client_secret,
+                data.surveymonkey_api_cache_duration,
+            )
         except InvalidClientError:
             validation.add(ValidationMessage(ValidationMessage.ERROR, u"Invalid client id or client secret"))
             return
 
-        self._validate_survey_name(validation, data, api_survey_monkey)
+        self.survey_id = self._validate_survey_name(validation, data.survey_name, api_survey_monkey)
+
+        self._get_survey_link(self.survey_id, validation, api_survey_monkey)
```

Comment:
```suggestion
        if  not survey_id:
            return
        self._get_survey_link(self.survey_id, validation, api_survey_monkey)
```

Reply:
return None, don't forget it. Changed.

Reply:
this is recommend when the method returns a value, it doesn't apply  for this case

Reply:
But lines above you are using an empty return statement, anywise I'll always require it.

Reply:
https://stackoverflow.com/questions/15300550/return-return-none-and-no-return-at-all